products/compute/api.yaml

```diff
@@ -7848,6 +7848,238 @@ objects:
+        required: true
+      - !ruby/object:Api::Type::String
+        name: 'displayName'
+        description: |
+          The display name of the security policy of the association.
+        output: true
+  - !ruby/object:Api::Resource
+    name: 'OrganizationSecurityPolicyRule'
+    min_version: beta
+    base_url: 'locations/global/securityPolicies/{{policy_id}}'
+    self_link: 'locations/global/securityPolicies/{{policy_id}}/getRule?priority={{priority}}'
+    create_url: 'locations/global/securityPolicies/{{policy_id}}/addRule?priority={{priority}}'
+    update_verb: :POST
+    update_url: 'locations/global/securityPolicies/{{policy_id}}/patchRule?priority={{priority}}'
+    delete_verb: :POST
+    delete_url: 'locations/global/securityPolicies/{{policy_id}}/removeRule?priority={{priority}}'
+    description: |
+      A rule for the OrganizationSecurityPolicy.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Creating firewall rules':
+          'https://cloud.google.com/vpc/docs/using-firewall-policies#create-rules'
+      api: 'https://cloud.google.com/compute/docs/reference/rest/beta/organizationSecurityPolicies/addRule'
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: policyId
+        description: |
+          The ID of the OrganizationSecurityPolicy this rule applies to.
+        required: true
+        input: true
+        url_param_only: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'description'
+        description: |
+          A description of the rule.
+      - !ruby/object:Api::Type::Integer
+        name: 'priority'
+        description: |
+          An integer indicating the priority of a rule in the list. The priority must be a value
+          between 0 and 2147483647. Rules are evaluated from highest to lowest priority where 0 is the
+          highest priority and 2147483647 is the lowest prority.
+        required: true
+        input: true
+      - !ruby/object:Api::Type::NestedObject
+        name: 'match'
+        description:
+          A match condition that incoming traffic is evaluated against. If it evaluates to true,
+          the corresponding 'action' is enforced.
+        required: true
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'description'
+            description: |
+              A description of the rule.
+          - !ruby/object:Api::Type::Enum
+            name: 'versionedExpr'
+            description: |
+              Preconfigured versioned expression. For organization security policy rules,
+              the only supported type is "FIREWALL".
+            values:
+              - :FIREWALL
+            default_value: :FIREWALL
+          - !ruby/object:Api::Type::NestedObject
+            name: 'config'
+            description:
+              The configuration options for matching the rule.
+            required: true
+            properties:
```

Comment:
Can one of these properties be set to required, or can we do an AtLeastOneOf on them?

Reply:
made config required (as it is) and exactly one of src or dest ip ranges (depends if its INGRESS or EGRESS rules. Also clarified in the documentation.